### ISSUE TITLE
IGNOREEOF doesn't work

### ISSUE THREAD

**Issue by OJFord** [xonsh/xonsh#1556]
I set `IGNOREEOF` during the config wizard, but it's not having any effect - I can leave xonsh with Ctrl-D.

```sh
$ grep 'IGNOREEOF' "$XDG_CONFIG_HOME/xonsh/config.json"
  "IGNOREEOF": true,
$ xonsh --version
('xonsh/0.4.4',)
$ uname -r
4.7.0-1-ARCH
```

**Comment by laerus** [comment]
this exists on current master as well and it's defo a bug. nice catch!

`$IGNOREEOF` doesn't affect `EOF` too

**Comment by scopatz** [comment]
Thanks for reporting @OJFord! Potential fix in #1557

**Comment by scopatz** [comment]
The fix for this has been merged.  Please feel free to reopen if needed.
